### PR TITLE
JST-70: Automate releases

### PR DESCRIPTION
**Pull request by grisha87**
This PR introduces `semantic-release` to the project.

As part of this PR we:

- separated a fast CI pipeline which is concerned about formatting, linting and unit tests
- separated a slow E2E and release pipeline which is there to ensure more testing before the automated release
- dropped tests on Node 19 as it's not supported anymore
- this PR requires from developers to follow the conventional commit message formatting
- Goth and Cypress tests will be runnable manually (triggering from the UI)

Thanks to these changes we gain:

- shorter feedback loops for PRs
- automated releases to GH, NPM including automated generation of release notes

Closes #173

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      - name: Perform regular checks
+        run: |
+          npm install
+          npm run format:check
+          npm run lint
+          npm run test:unit
+          npm run build
```
Comment (mgordel) [inline .github/workflows/ci.yml:41]: Maybe we should do a build before units first? if TSC doesn't pass, there's no need to run tests..

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+on:
+  push:
+    branches:
+      # Regular release channels
+      - master
+      - next
+      - beta
+      - alpha
+      # Support, hotfix branches like: 1.0.x or 1.x
+      - "+([0-9])?(.{+([0-9]),x}).x"
```
Comment (mgordel) [inline .github/workflows/release.yml:12]: is this regex correct?
I'm not sure but I tested on validator and I think it should be: `([0-9])+(.{+([0-9]),x})?.x`

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      - name: Perform regular checks
+        run: |
+          npm install
+          npm run format:check
+          npm run lint
+          npm run test:unit
```
Comment (mgordel) [inline .github/workflows/release.yml:44]: as above .. ?

**Review by mgordel — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (grisha87) [reply .github/workflows/ci.yml:41]: As per Discord -> test:unit will force tsc due to `ts-jest`, so transpilation errors will fail the build early.

File: .github/workflows/release.yml (hunk 2 above)
Comment (grisha87) [reply .github/workflows/release.yml]: Nice catch! That's actually a pattern for `micromatch`, so I patched the regex used in the workflow files to a valid reg-exp.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (mgordel) [reply .github/workflows/ci.yml:41]: You're right

**Pull request merged**